### PR TITLE
Added stripped DefaultComponents for Lambda Extension

### PR DESCRIPTION
**Pull request by AsakerMohd**
**Description:** Added a new default component function under the `defaultcomponents` package. This would be imported by the AOC Lambda Extension under aws-otel-lambda to build a single binary containing inline AOC into Lambda Extension. 
Refer to https://quip-amazon.com/YT0QAfgSHDfY/Inline-AOC-into-extension for details.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** This was tested by importing it into the main function instead of the existing default components. Running the `make amd64-build` created a binary that has a size of ~28 MB compared to the ~70 MB when built using the existing default components.

### CODE REVIEW

File: pkg/defaultcomponents-lambda/defaults.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+ package defaultcomponents // import "aws-observability.io/collector/defaultcomponents
+
+ import (
+	 "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter"
+	 "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter"
+	 "go.opentelemetry.io/collector/component"
+	 "go.opentelemetry.io/collector/component/componenterror"
+	 "go.opentelemetry.io/collector/exporter/loggingexporter"
+	 "go.opentelemetry.io/collector/exporter/otlphttpexporter"
```
Comment (wangzlei) [inline pkg/defaultcomponents-lambda/defaults.go:24]: We also need otlpexporter

**Comment by wangzlei** [comment]
How does aws-otel-lambda consume it?

**Comment by AsakerMohd** [comment]
> How does aws-otel-lambda consume it?

We will first need to publish the AOC repo to GitHub. After that, we would be able to consume it like any other import:
`import "github.com/aoc-otel-collector/pkg/defaultcomponent"`

File: pkg/defaultcomponents-lambda/defaults.go (hunk 1 above)
Comment (AsakerMohd) [reply pkg/defaultcomponents-lambda/defaults.go]: resolved

**Review by wangzlei — APPROVED** [review]

**Review by wangzlei — APPROVED** [review]

**Review by wangzlei — APPROVED** [review]

**Review by wangzlei — APPROVED** [review]
looks good, thanks

File: pkg/defaultcomponents-lambda/defaults.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+ package defaultcomponents // import "aws-observability.io/collector/defaultcomponents
```
Comment (mxiamxia) [inline pkg/defaultcomponents-lambda/defaults.go:16]: It duplicated with the regular `defaultcomponents` package namespace. do you want to rename it to lambdacomponents?

Comment (AsakerMohd) [reply pkg/defaultcomponents-lambda/defaults.go]: Done.

File: pkg/lambdacomponents/defaults.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		 errs = append(errs, err)
+	 }
+ 
+	 exporters, err := component.MakeExporterFactoryMap(
+		 awsxrayexporter.NewFactory(),
+		 awsemfexporter.NewFactory(),
+		 prometheusexporter.NewFactory(),
+		 loggingexporter.NewFactory(),
+		 otlpexporter.NewFactory(),
+		 otlphttpexporter.NewFactory(),
```
Comment (mxiamxia) [inline pkg/lambdacomponents/defaults.go:51]: @wangzlei Do we really need to support all these exporters in Lambda, especially `prometheusexporter`, how can prometheus scrape metrics from Lambda extension?

**Review by mxiamxia — COMMENTED** [review]
Can we also add `go.mod` in this PR? so we know we will make this as a go module for Lambda extension to use. Thanks

File: pkg/lambdacomponents/defaults.go (hunk 3 above)
Comment (AsakerMohd) [reply pkg/lambdacomponents/defaults.go:51]: I know otlpexporter is. I'm not sure about the others.

**Comment by AsakerMohd** [comment]
> Can we also add `go.mod` in this PR? so we know we will make this as a go module for Lambda extension to use. Thanks

Yeah makes sense. I'll add one.

**Review by mxiamxia — APPROVED** [review]

**Pull request merged**